### PR TITLE
Expire deprecations in backends

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

~Will add release notes.~ Done!

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: doc/api/next_api_changes/removals/22516-OG.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Removal of deprecations in ``backends``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The parameter ``resize_callback`` is removed from the TK backend. Instead,
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/22516-OG.rst:4]:
```suggestion
The parameter ``resize_callback`` is removed from the Tk backend. Instead,
```

File: doc/api/next_api_changes/removals/22516-OG.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+use ``get_tk_widget().bind('<Configure>', ..., True)``.
+
+The ``get_content_extents()`` and ``tostring_rgba_minimized()`` methods are
+removed from the Agg backend with no replacements.
+
+The parameter ``dpi`` is removed from ``_print_ps()`` in the PS backend and
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/22516-OG.rst:10]: `_print_ps` is private; it's `print_ps` and `print_eps` that have the parameter removed.

**Comment by oscargus** [comment]
Thanks! I fixed those.

**Review by tacaswell — APPROVED** [review]

File: doc/api/next_api_changes/removals/22516-OG.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Removal of deprecations in ``backends``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The parameter ``resize_callback`` is removed from the Tk backend. Instead,
+use ``get_tk_widget().bind('<Configure>', ..., True)``.
+
+The ``get_content_extents()`` and ``tostring_rgba_minimized()`` methods are
+removed from the Agg backend with no replacements.
+
+The parameter ``dpi`` is removed from ``print_ps()`` in the PS backend and
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/22516-OG.rst:10]: Missing `print_eps` as well.

Comment (oscargus) [reply doc/api/next_api_changes/removals/22516-OG.rst:10]: OK! Fixed!

**Review by timhoffm — APPROVED** [review]

**Pull request merged**